extensions/HyperVExtension/src/HyperVExtension/Services/HyperVManager.cs

```diff
@@ -367,8 +367,23 @@ public bool RemoveVirtualMachine(Guid vmId)
                 return false;
             }
 
-            var virtualMachine = _hyperVVirtualMachineFactory(vmObject);
-            return virtualMachine.IsDeleted;
+            var psObjectHelper = new PsObjectHelper(vmObject);
+
+            // Check if the Hyper-V virtual machine object returned to use has its IsDeleted status set to true
+            if (psObjectHelper.MemberNameToValue<bool>("IsDeleted"))
+            {
+                return true;
+            }
+
+            // Unfortunately sometimes the IsDeleted property can be set to false and dynamically
+            // update to true later on. There is no documentation for the Hyper-V objects, or best practices for the Hyper-V PowerShell
+            // cmdlets, so we will do an additional check by attempting to get the VM with the Get-VM cmdlet. If there are no results
+            // we can say the VM has been deleted.
+            result = _powerShellService.Execute(GetVMCommandLineStatement(vmId), PipeType.None);

```

Comment:
I have it because its quicker to check the variable, versus doing another powershell command. So in cases where the variable is actually true then we don't need to do another powershell execution.